vyper/contracts/libs/FulfillmentApplier.vy

```diff
@@ -0,0 +1,545 @@
+    consideration: ReceivedItem = orderToExecute.receivedItems[potentialCandidate.itemIndex]
+
+    # Create the received item.
+    receivedItem = ReceivedItem({
+        itemType: consideration.itemType,
+        token: consideration.token,
+        identifier: consideration.identifier,
+        amount: consideration.amount,
+        recipient: consideration.recipient
+    })
+
+    # Zero out amount on original offerItem to indicate it is spent
+    consideration.amount = 0
+
+    # Loop through the consideration components and validate
+    # their fulfillment.
+    for component in considerationComponents:
+        # Get the order index and item index of the consideration component.
+        potentialCandidate.orderIndex = component.orderIndex
+        potentialCandidate.itemIndex = component.itemIndex
+
+        # Ensure that the order index is not out of range.
+        potentialCandidate.invalidFulfillment = potentialCandidate.orderIndex >= len(ordersToExecute)
+        # Break if invalid
+        if potentialCandidate.invalidFulfillment:
+            break
+
+        # Get the order based on consideration components order index.
+        orderToExecute = ordersToExecute[potentialCandidate.orderIndex]
+        # Confirm this is a fulfilled order.
+        if orderToExecute.numerator != 0:
+            # Ensure that the item index is not out of range.
+            potentialCandidate.invalidFulfillment = \
+                (potentialCandidate.itemIndex >= len(orderToExecute.receivedItems))
+            # Break if invalid
+            if potentialCandidate.invalidFulfillment:
+                break
+
+            # Retrieve relevant item using item index.
+            consideration = orderToExecute.receivedItems[potentialCandidate.itemIndex]
+            # Updating Received Item Amount
+            receivedItem.amount = receivedItem.amount + consideration.amount
+            # Zero out amount on original consideration item to indicate it is spent
+            consideration.amount = 0
+            # Ensure the indicated consideration item matches original item.
+            potentialCandidate.invalidFulfillment = self._checkMatchingConsideration(
+                consideration,
+                receivedItem
+            )
+
+    # Revert if an order/item was out of range or was not aggregatable.
+    assert not potentialCandidate.invalidFulfillment, "invalid fulfillment component data"
+    return receivedItem
+
+@internal
+@pure
+def _checkMatchingOffer(
+    orderToExecute: OrderToExecute,
+    offer: SpentItem,
+    execution: Execution
+) -> bool:
+    """
+    @dev Internal pure function to check the indicated offer item matches original item.
+
+    @param orderToExecute  The order to compare.
+    @param offer The offer to compare
+    @param execution  The aggregated offer item
+
+    @return invalidFulfillment A boolean indicating whether the fulfillment is invalid.
+    """
+    return \
```

Comment:
ditto for comments on `_checkMatchingConsideration`. `amount` is missing(?) and, maybe clearer to return the contrapositive.

Reply:
changed to contrapositive